qiskit/utils/deprecation.py

```diff
@@ -77,3 +92,73 @@ def _rename_kwargs(func_name, kwargs, kwarg_map, category: Type[Warning] = Depre
                 )
 
                 kwargs[new_arg] = kwargs.pop(old_arg)
+
+
+def _extend_docstring(func, version, kwarg_map):
+    """kwarg_map[None] means message in no kwarg, and it will be
+    append to the long desscription"""
+    docstr = func.__doc__
+    if docstr:
+        docstr_lines = docstr.expandtabs().splitlines()
+    else:
+        docstr_lines = ["DEPRECATED"]
+
+    indent = 1000
+    first_empty_line = None
+    for line_no, line in enumerate(docstr_lines[1:], start=1):
+        stripped = line.lstrip()
+        if stripped:
+            indent = min(indent, len(line) - len(stripped))
+        else:
+            if first_empty_line is None:
+                first_empty_line = line_no
+    if first_empty_line is None:
+        first_empty_line = len(docstr_lines)
+    spaces = ""
+    if indent != 1000:
+        spaces = " " * indent
```

Comment:
If this is copied / modified from PEP 257 we should include a comment with a link, just for the context. I think we're fine from a copyright perspective, since that document places itself in the public domain.

Reply:
adding the comment in fbc2d74925293b7c63df7586ba89765d7ccb318b